website/docs/migrating-from-v1-to-v2.md

```diff
@@ -492,3 +494,154 @@ You might want to refer to our migration PRs for [Create React App](https://gith
 ## Support
 
 For any questions, you can ask in the [`#docusaurus-1-to-2-migration` Discord channel](https://discordapp.com/invite/kYaNd6V). Feel free to tag us ([@yangshun](https://github.com/yangshun) and [@endiliey](https://github.com/endiliey)) in any migration PRs if you would like us to have a look.
+
+## Versioned Site
+
+> :warning: _This section is a work in progress._
```

Comment:
We need to think about the [version page](https://github.com/facebook/docusaurus/blob/master/website/src/pages/versions.js). I mean that it immediately exists if versioning is used, instead of having users manually create it. WDYT?

Reply:
Hmmm, but 'versions' page route is handled by `@docusaurus/plugin-content-pages` and not `@docusaurus/plugin-content-docs`. 

I'm afraid that if versions page route is handled by docs plugin, this means they cannot customize it.
In docusaurus v1 it was possible because all functionality is too coupled, but in v2 we might allow just docs plugin with even different theme package

But I think we can just mention in the docs they can copy our versions page 😆

Reply:
Oh yes, I would then create a command that would copy the template for the version page :)

By the way, I would change the display of the current version in navbar to something like this
![image](https://user-images.githubusercontent.com/4408379/69532352-a6ff3980-0f86-11ea-9617-a492a847cf4e.png)

That is better?

Reply:
This is just a prototype, in the current implementation, this will also be a separate menu item (yes, styling), but then on mobiles, version will not be visibled. Therefore, we need to do this as part of the logo itself, but for this we need to swizzle NavBar, otherwise how to add custom HTML to the logo item?

Reply:
Maybe we can just make it not compelx

We create new API
```js
// docusaurus.config.js
module.exports = {
  themeConfig: {
    navbar: {
      title: 'Site Title',
      logo: {
        alt: 'Site Logo',
        src: 'img/logo.svg',
+     defaultVersionShown: '2.0.0-alpha.36'
      },
    }
}
```

When on `/docs/next/`, or any other `docs` route, the layout is aware which version it is in
https://github.com/facebook/docusaurus/blob/9b4715a70da1d3acb6fca963dc7f6bb6b6621e5e/packages/docusaurus-theme-classic/src/theme/Layout/index.js#L34-L35

So we can pass it to navbar and display the current docs version we're reading in
It's like this but we can move it to top (near logo) (ignore the bad css here, i used chrome to hack this out)
![image](https://user-images.githubusercontent.com/17883920/69535836-5223e900-0faf-11ea-96e3-00cc86623034.png)


If they are in '/pages/about', '/blog/xxx', we just show the defaultVersion,
Clicking the logo will link to `/versions` if `defaultVersionShown` is defined

https://github.com/facebook/docusaurus/blob/9b4715a70da1d3acb6fca963dc7f6bb6b6621e5e/packages/docusaurus-theme-classic/src/theme/Navbar/index.js#L103-L113

WDYT

Reply:
But need to make sure this looks good on mobile too :(

Reply:
So we want to display the version for all users ([like React website](https://github.com/reactjs/reactjs.org/blob/master/src/site-constants.js#L11))? I would do then not bind the version to the header. Something like this:

```
themeConfig: {
	...
	version: {
		current:
		showInNavbar:
    }
	...
}
```

Or simpler - we can take the [first version element ](https://github.com/facebook/docusaurus/blob/master/website/src/pages/versions.js#L21)as on the versions page

Reply:
yeah ok maybe lets not bind it to header. its kinda weird because only docs is versioned.

I think at the end of the day, we want something like in infima, using the dropdown https://infima-dev.netlify.com/

lets not do anything for now. Its too coupled with infima. I believe @yangshun has his plan for how the site should look like.